### PR TITLE
feat: expose axios_config

### PR DESCRIPTION
**Pull request by mattrothenberg**
This PR exposes a new `axios_config` parameter that can be used to tailor the request made as part of the `http` backend.

Resolves https://github.com/githubocto/flat/issues/53

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,8 +101,26 @@ For example, if this field is set to `Bearer abc123` then the following header i
...
+An example `axios_config` might look thusly if you were interested in hitting GitHub's GraphQL API 👇
+
+```json
+{
+  "method": "post",
+  "data": {
+    "query": "query { repository(owner:\"octocat\", name:\"Hello-World\") { issues(last:20, states:CLOSED) { edges { node { title url labels(first:5) { edges { node { name } } } } } } } }"
+  }
+}
+```
```
Comment (irealva) [inline README.md:123]: Totally optional: we could add a line somewhat like this:

We advise escaping double quotes `\"` in your JSON file

File: README.md [hunk 2]
```diff
@@ -101,8 +101,26 @@ For example, if this field is set to `Bearer abc123` then the following header i
...
+
+Under the hood, the `http` backend uses [Axios](https://github.com/axios/axios) for data fetching. By default, Flat assumes you're interested in using the `GET` method to fetch data, but if you'd like to `POST` (e.g., sending a GraphQL query), the `axios_config` option allows you to override this behavior.
+
+Specifically, the `axios_config` parameter should reflect a relative path to a `.json` file in your repository. This JSON file should mirror the shape of [Axios' request config parameters](https://github.com/axios/axios#request-config), with a few notable exceptions.
+
+- `url` and `baseURL` will both be ignored, as the `http_url` specified above will take precedence.
+- `headers` will be merged in with the authorization header described by the `authorization` parameter above. Please do not put secret keys here, as they will be stored in plain text!
+- All `function` parameters will be ignored (e.g., `transformRequest`).
+
+An example `axios_config` might look thusly if you were interested in hitting GitHub's GraphQL API 👇
```
Comment (irealva) [inline README.md:114]: Also optional: 

"Here's a [demo](https://github.com/githubocto/flat-demo-graphql)"

Comment (mattrothenberg) [reply README.md:114]: @irealva Feel free to make those updates!!!

File: README.md [hunk 3]
```diff
@@ -101,8 +101,26 @@ For example, if this field is set to `Bearer abc123` then the following header i
...
+#### `axios_config` (optional)
+
+Under the hood, the `http` backend uses [Axios](https://github.com/axios/axios) for data fetching. By default, Flat assumes you're interested in using the `GET` method to fetch data, but if you'd like to `POST` (e.g., sending a GraphQL query), the `axios_config` option allows you to override this behavior.
+
+Specifically, the `axios_config` parameter should reflect a relative path to a `.json` file in your repository. This JSON file should mirror the shape of [Axios' request config parameters](https://github.com/axios/axios#request-config), with a few notable exceptions.
+
+- `url` and `baseURL` will both be ignored, as the `http_url` specified above will take precedence.
+- `headers` will be merged in with the authorization header described by the `authorization` parameter above. Please do not put secret keys here, as they will be stored in plain text!
+- All `function` parameters will be ignored (e.g., `transformRequest`).
+
```
Comment (irealva) [inline README.md:113]: Should we mention the thing about responseType always will be set to 'stream' in the background?

Comment (mattrothenberg) [reply README.md:113]: Definitely, I completely forgot

**Comment by irealva** [comment]
Squashed commits

**Pull request merged**